packages/search-token-server/server.spec.ts

```diff
@@ -0,0 +1,187 @@
+jest.mock('@coveord/platform-client');
+jest.mock('./middlewares/searchToken', () => {
+  const originalMiddleware = jest.requireActual('./middlewares/searchToken');
+  const oldMiddleware = originalMiddleware.ensureTokenGenerated;
+
+  originalMiddleware.ensureTokenGenerated = function (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) {
+    mockedMiddleware(req, res, next);
+    oldMiddleware(req, res, next);
+  };
+
+  return originalMiddleware;
+});
+
+import {mocked} from 'ts-jest/utils';
+import {agent} from 'supertest';
+import app from './app';
+import {Request, Response, NextFunction} from 'express';
+import PlatformClient, {RestUserIdType} from '@coveord/platform-client';
+
+const mockedPlatformClient = mocked(PlatformClient);
+const mockedCreateToken = jest.fn();
+const mockedMiddleware = jest.fn();
+
+describe('server', () => {
+  const OLD_ENV = process.env;
+
+  const setInvalidEnvironmentVariables = () => {
+    process.env.ORGANIZATION_ID = undefined;
+    process.env.API_KEY = undefined;
+    process.env.USER_EMAIL = undefined;
+  };
+
+  const setValidEnvironmentVariables = () => {
+    process.env.PLATFORM_URL = 'https://platform.cloud.coveo.com';
+    process.env.ORGANIZATION_ID = 'my-org';
+    process.env.API_KEY = 'xxx';
+    process.env.USER_EMAIL = 'bob@coveo.com';
+  };
+
+  const triggerAMiddlewareError = () => {
+    mockedMiddleware.mockImplementationOnce(
+      (_req: Request, _res: Response, next: NextFunction) => {
+        const err: Error = new Error();
+        next(err);
+      }
+    );
+  };
+  const resetAppLocals = () => {
```

Comment:
I'm really splitting hair here, but stay consistent in your breakline :P 
```suggestion
  };

  const resetAppLocals = () => {
```